main.py

```diff
@@ -46,19 +62,80 @@ def handler(event, context):
 
 def configure_confluent_kafka_consumer(event, args):
     if 'AWS_PROFILE' in os.environ:
-        boto3.setup_default_session(profile_name=args.aws_profile,
-                                    region_name=args.aws_region)
+     boto3.setup_default_session(profile_name=args.aws_profile,
+     region_name=args.aws_region)
 
     if logger.isEnabledFor(logging.DEBUG):
         # Log everything from boto3
         boto3.set_stream_logger()
-        logger.debug(f"Using boto3 {boto3.__version__}")
+    logger.debug(f"Using boto3 {boto3.__version__}")
 
     logger.debug(event)
 
     message = json.loads(event['Records'][0]['Sns']['Message'])
     logger.debug(message)
 
+    private_ip = message['detail']['containers'][0]['networkInterfaces'][0]['privateIpv4Address']
+    logger.debug(private_ip)
+
+    payload = {
+        "config": {
+            "connector.name": args.connector_name,
+            "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+            "tasks.max": args.tasks_max,
+            "topics": args.topics,
+            "flush.size": args.flush_size,
+            "s3.region": args.aws_region,
+            "s3.bucket.name": args.s3_bucket_name,
+            "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+            "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
+            "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
+            "partitioner.class": "io.confluent.connect.storage.partitioner.DefaultPartitioner",
+        }
+    }
+
+
+# Get a list of existing connectors
+
+    response = requests.get(f"http://{private_ip}:{args.port}/connectors")
+    existing_connectors = response.content.decode("utf-8")
+    logger.debug(existing_connectors)
+
+    # PUT payload if connectors do exist this updates existing connector
+
+    if args.connector_name in existing_connectors:
+        logger.debug("update connector [PUT]")
+
+        payload = {
+            "config": {
+                "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+                "tasks.max": args.tasks_max,
+                "topics": args.topics,
+                "flush_size": args.flush_size,
```

Comment:
I'm going to have to confirm in the docs, but I'm pretty sure there should be more here, at least the S3 bucket and region